pyfr/plugins/fluidforce.py

```diff
@@ -122,6 +164,29 @@ def __call__(self, intg):
             # Do the quadrature
             f += np.einsum('i...,ij,jik', qwts, p, norms)
 
+            if self.forcetype == 'viscous':
+                # Get operator and J^-T matrix
+                m4 = self._m4[etype]
+                m5 = self._m5[etype, fidx]
+                rcpjact = self._rcpjact[etype, fidx]
+
+                # Transformed gradient at solution points
+                tduupts = np.dot(m4, uupts.reshape(nupts, -1))
+                tduupts = tduupts.reshape(ndims, nupts, nvars, -1)
+
+                # Physical gradient at solution points
+                duupts = np.einsum('ijkl,jkml->ikml', rcpjact, tduupts)
+
+                # Interpolate gradient to flux points
+                dufpts = np.dot(m5, duupts.reshape(ndims*nupts, -1))
```

Comment:
I think we should be doing multiple applications of `m0` here.  It avoids the need for `m5` and the inefficiencies associated with it.